File: backend/app/schemas/resume.py
from pydantic import BaseModel, Field


class Education(BaseModel):
    institution: str | None = None
    degree: str | None = None
    field: str | None = None
    cgpa: float | None = None
    start_year: int | None = None
    end_year: int | None = None


class Experience(BaseModel):
    company: str | None = None
    role: str | None = None
    duration: str | None = None
    description: str | None = None


class Project(BaseModel):
    title: str | None = None
    description: str | None = None
    technologies: list[str] = Field(default_factory=list)


class ResumeSchema(BaseModel):
    name: str | None = None
    email: str | None = None
    phone: str | None = None
    github: str | None = None
    linkedin: str | None = None
    summary: str | None = None

    skills: list[str] = Field(default_factory=list)
    education: list[Education] = Field(default_factory=list)
    experience: list[Experience] = Field(default_factory=list)
    projects: list[Project] = Field(default_factory=list)

File: frontend/src/types/resume.ts
export interface Education {
  institution: string | null;
  degree: string | null;
  field: string | null;
  cgpa: number | null;
  start_year: number | null;
  end_year: number | null;
}

export interface Experience {
  company: string | null;
  role: string | null;
  duration: string | null;
  description: string | null;
}

export interface Project {
  title: string | null;
  description: string | null;
  technologies: string[];
}

export interface Resume {
  name: string | null;
  email: string | null;
  phone: string | null;
  github: string | null;
  linkedin: string | null;
  summary: string | null;

  skills: string[];
  education: Education[];
  experience: Experience[];
  projects: Project[];
}

export interface ResumeAnalysis {
  strengths: string[];
  weaknesses: string[];
  skill_gaps: string[];
  recommended_topics: string[];
}